low-level/files/opt/seagate/sspl/setup/sspl_post_install.py

```diff
@@ -207,15 +207,43 @@ def create_sspl_conf(self):
             shutil.copyfile("%s/conf/sspl.conf.%s.yaml" % (consts.SSPL_BASE_DIR,
                 self.product), consts.file_store_config_path)
 
+    def validate_user_created(self):
+        max_retry = 3
+        for i in range(max_retry):
+            sspl_uid = Utility.get_uid(consts.USER)
+            if sspl_uid != -1:
+                break
+            else:
+                if i == (max_retry-1):
+                    msg = "No user found with name : %s" % (consts.USER)
+                    logger.error(msg)
+                    raise SetupError(errno.EINVAL, msg)
+                time.sleep(1)
+
+    def validate_group_created(self):
+        max_retry = 3
+        for i in range(max_retry):
+            sspl_gid = Utility.get_gid(consts.USER)
+            if sspl_gid != -1:
+                break
+            else:
+                if i == (max_retry-1):
+                    msg = "No group found with name : %s" % (consts.USER)
+                    logger.error(msg)
+                    raise SetupError(errno.EINVAL, msg)
+                time.sleep(1)
+
     def create_user(self):
         """Add sspl-ll user and validate user creation."""
-        os.system("/usr/sbin/useradd -r %s -s /sbin/nologin \
-            -c 'User account to run the %s service'" % (consts.USER, consts.USER))
-        usernames = [x[0] for x in pwd.getpwall()]
-        if consts.USER not in usernames:
-            msg = "User %s doesn't exit. Please add user." % (consts.USER)
-            logger.error(msg)
-            raise SetupError(errno.EINVAL, msg)
+        command = f"/usr/sbin/useradd -r {consts.USER} -s /sbin/nologin \
+                  -c 'User account to run the {consts.USER} service'"
+        _, error, rc = SimpleProcess(command).run()
+        if rc != 0:
+            logger.error(f"Failed to create sspl user. ERROR: {error}")
+            raise SetupError(rc, error, command)
```

Comment:
if we raise exception here, even validation retries wont happen through line 245,246, rt ?

Reply:
Yes